tests/system/src/test/java/test/com/sun/glass/ui/win/WinTextRangeProviderTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.com.sun.glass.ui.win;
+
+import com.sun.javafx.PlatformUtil;
+import com.sun.glass.ui.win.WinTextRangeProviderShim;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.stream.Stream;
+
+import test.util.Util;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+public class WinTextRangeProviderTest {
+
+    private static final CountDownLatch startupLatch = new CountDownLatch(1);
+
+    @BeforeAll
+    static void initFX() throws Exception {
+        assumeTrue(PlatformUtil.isWindows());
+        Util.startup(startupLatch, () -> startupLatch.countDown());
+    }
+
+    @AfterAll
+    static void shutdown() {
+        Util.shutdown();
```

Comment:
This fails on Mac and Linux because the initialization (and consequently all tests) are skipped, but the call to `Util.shutdown` is not skipped. You will either need to also add `assumeTrue(PlatformUtil.isWindows());` here or else move the `assumeTrue` to the test method.

Reply:
is there a good way to skip the whole file entirely, e.g. via junit annotation?

Reply:
maybe
https://stackoverflow.com/questions/26115139/junit-ignore-or-skip-entire-test-class
?

Reply:
We still want to use `assumeTrue` as the pattern (for flexibility), so the question is whether there is a better place besides a method annotated with `@BeforeAll` to put it (which also means duplicating the assumption in the `@AfterAll` method). I'm not aware of one, but maybe someone else is?

Reply:
Junit 5 has an EnabledOnOs annotation https://junit.org/junit5/docs/5.2.0/api/org/junit/jupiter/api/condition/EnabledOnOs.html

Reply:
> EnabledOnOs

thank you @Siedlerchr for this info.
I wonder what criteria are set for determining the OS value, and do they match jfx ones?

Reply:
They just read the `os.name` with `System.getProperty("os.name")`
https://github.com/junit-team/junit5/blob/db47616ab4ccf38ff63e8bff41050d5102c9ff15/junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/OS.java#L110

Reply:
the logic **is** different, but both variants should be ok for testing purposes.

for example,
junit: mac = toLowercase(ENGLISH).contains("mac")
jfx: mac = startsWith("Mac")

Reply:
I prefer to stick with assumptions since that's what we use everywhere else. Also, it gives us more control.

So while we _could_ do something else in the future, this PR isn't the place to do it.